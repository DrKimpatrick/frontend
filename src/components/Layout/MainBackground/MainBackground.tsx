import React, { FC } from 'react';
import Image from '../../../assets/images/motif-image.png';
import './style.scss';

const MainBackground: FC = () => {
  return (
    <div className="mainBackground">
      <img src={Image} alt="" />
    </div>
  );
};

export default MainBackground;
