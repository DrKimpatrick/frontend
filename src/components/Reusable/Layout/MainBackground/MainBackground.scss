.main {
  padding-bottom: 0px !important;
}

.mainBackground {
  position: relative;
  width: 100%;
  height: auto;
  z-index: -200;
  margin-top: -200px;
  &::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    z-index: 100;
    background-color: #fbfbfb00;
    width: 100%;
    height: 100%;
  }
  img {
    width: 100%;
    opacity: 0.2;
    mix-blend-mode: luminosity;
  }
}
