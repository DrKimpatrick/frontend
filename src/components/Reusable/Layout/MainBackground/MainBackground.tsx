import React, { FC } from 'react';
import Image from 'assets/images/motif-image.png';
import './style.scss';

interface Props {
  allowImageColor?: boolean;
}
const MainBackground: FC<Props> = props => {
  const { allowImageColor } = props;

  return (
    <div className="mainBackground">
      <img
        src={Image}
        alt=""
        style={
          allowImageColor
            ? { backgroundColor: '#c4bdbd' }
            : { backgroundColor: 'none' }
        }
      />
    </div>
  );
};

export default MainBackground;
