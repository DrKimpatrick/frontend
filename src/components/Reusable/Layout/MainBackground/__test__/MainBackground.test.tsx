import React from 'react';
import ReactDom from 'react-dom';
import MainBackground from '../MainBackground';
import renderer from 'react-test-renderer';

describe('`MainBackground`', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<MainBackground />, div);
  });

  it('should render correctly', () => {
    const tree = renderer.create(<MainBackground />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
