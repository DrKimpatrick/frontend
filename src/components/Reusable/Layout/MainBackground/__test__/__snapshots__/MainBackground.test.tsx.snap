// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`MainBackground\` should render correctly 1`] = `
<div
  className="mainBackground"
>
  <img
    alt=""
    src="motif-image.png"
    style={
      Object {
        "backgroundColor": "none",
      }
    }
  />
</div>
`;
